### ISSUE TITLE
How to debug `ranger` being completely frozen/unresponsive

### ISSUE THREAD

**Issue by jamesbraza** [ranger/ranger#3017]
#### Runtime Environment

- Operating system: macOS Sonoma v14.5, with M3 chip
- ranger version: ranger 1.9.3
- Terminal emulator and version: iTerm2 build 3.5.5
- Python version: Python version: 3.12.7 (main, Oct  1 2024, 02:05:46) [Clang 15.0.0 (clang-1500.3.9.4)]
- Ranger version/commit: `brew install ranger`
- Locale: California, US

#### Current Behavior

I run `ranger` and it's completely unresponsive, it's completely frozen and unresponsive.

#### Expected Behavior

`ranger` not to be frozen..

#### Context

Here is my `~/.config/ranger/rc.conf`:

```none
# Disable displaying line numbers in main column.
# Possible values: false, absolute, relative.
set line_numbers absolute

# SEE: https://github.com/alexanderjeurissen/ranger_devicons#install-instructions
default_linemode devicons

# SEE: https://github.com/ranger/ranger/wiki/Image-Previews
set preview_images true
set preview_images_method iterm2
```

And I ran `ranger --copy-config=scope` with `brew install imagemagick poppler`

#### Possible Solutions

I don't know of any

#### Steps to reproduce

Run `ranger`

#### Traceback

No crash

**Comment by toonn** [comment]
I realize you might've moved on by this point but for posterity's sake.

I don't have any satisfying answer to this. Most likely culprit is pretty much always previews. ImageMagick in particular is a pretty big offender but Pandoc's another.

What could be a solution is to do more things asynchronously, that way slow previews just don't show up until they're done without slowing down Ranger. It's not a simple change though so I wouldn't hold my breath.

**Comment by jamesbraza** [comment]
Hi @toonn actually I am still interested in this, I like `ranger`.

> Most likely culprit is pretty much always previews. ImageMagick in particular is a pretty big offender but Pandoc's another.

Okay, trying with the below `~/.config/ranger/rc.conf`, and then closing/reopening my Iterm2 3.5.10, running `ranger` with `ranger==1.9.4`:

```none
# Disable displaying line numbers in main column.
# Possible values: false, absolute, relative.
set line_numbers absolute

# SEE: https://github.com/alexanderjeurissen/ranger_devicons#install-instructions
default_linemode devicons
```

I still get the hang, even with no ImageMagick. Any ideas?

**Comment by toonn** [comment]
Does the same problem happen when you run `ranger --clean`?

Have you tried disabling previews entirely?

**Comment by jamesbraza** [comment]
Yes that actually worked, `ranger --clean` is no longer frozen. I also tried using `ranger --debug` but nothing illuminating showed up.

I did a bunch of root causing, and found out commenting `export RANGER_LOAD_DEFAULT_RC=FALSE` had the same effect. So I believe was something in my `~/.config/ranger/scope.sh`.

Regardless, I did:

1. `sudo rm -r ~/.config/ranger`
1. `brew reinstall ranger`
1. `ranger --copy-config=rc`
1. Added `export RANGER_LOAD_DEFAULT_RC=FALSE` to `~/.zshenv`
1. Re-cloned the plugin https://github.com/alexanderjeurissen/ranger_devicons
1. Re-configured the above `rc.conf` overrides from [the OP](https://github.com/ranger/ranger/issues/3017#issue-2567510492)
 
Now my ranger is operational again. I don't know what happened.

Anyways, thanks again @toonn for your help. Sorry we didn't get a clearer bugfix from this, but
